language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -0,0 +1,229 @@
+
+                Position position = new Position(startLine, startChar);
+
+                if (parameterSymbols.size() <= i || parameterSymbols.get(i).getName().isEmpty()) {
+                    continue;
+                }
+                String label = parameterSymbols.get(i).getName().get();
+
+                InlayHint inlayHint = new InlayHint(position, Either.forLeft(label + ": "));
+                inlayHints.add(inlayHint);
+            }
+        }
+        return inlayHints;
+    }
+
+    private static List<ParameterSymbol> getParameterSymbols(InlayHintContext context,
+                                                             NonTerminalNode node) {
+        if (node.kind() == SyntaxKind.METHOD_CALL) {
+            MethodCallExpressionNode methodCallExpressionNode = (MethodCallExpressionNode) node;
+            Optional<FunctionTypeSymbol> libFunctions = TypeResolverUtil
+                    .findMethodInLangLibFunctions(methodCallExpressionNode, context);
+            if (libFunctions.isEmpty() || libFunctions.get().params().isEmpty()) {
+                return Collections.emptyList();
+            }
+            // Since the method is a lang-lib function, skip the first parameter
+            return libFunctions.get().params().get().stream().skip(1).collect(Collectors.toList());
+        } else {
+            FunctionCallExpressionNode functionCallExpressionNode = (FunctionCallExpressionNode) node;
+            return getFunctionCallParameterSymbols(context, functionCallExpressionNode);
+        }
+    }
+
+    private static List<ParameterSymbol> getFunctionCallParameterSymbols(InlayHintContext context,
+                                                                         FunctionCallExpressionNode node) {
+        LineRange lineRange = node.lineRange();
+        List<Symbol> visibleSymbols = context.visibleSymbols(new Position(
+                lineRange.startLine().line(), lineRange.startLine().offset()));
+
+        List<ParameterSymbol> parameterSymbols = visibleSymbols.stream()
+                .filter(symbol -> symbol.kind() == SymbolKind.FUNCTION)
+                .map(symbol -> (FunctionSymbol) symbol)
+                .filter(functionSymbol -> functionSymbol.getName().get().equals(
+                        node.functionName().toString().strip()))
+                .map(functionSymbol -> SymbolUtil.getTypeDescriptor(functionSymbol).get())
+                .filter(typeDescriptor -> typeDescriptor instanceof FunctionTypeSymbol)
+                .map(typeDescriptor -> (FunctionTypeSymbol) typeDescriptor)
+                .filter(functionTypeSymbol -> functionTypeSymbol.params().isPresent())
+                .map(functionTypeSymbol -> functionTypeSymbol.params().get())
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+
+        for (Node child : node.children()) {
+            if (child.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
```

Comment:
Let's remove this for loop. Check and find if the qualified name of the function call expression node is qualified name. Based on that condition we can decide to how to find the visible symbols.